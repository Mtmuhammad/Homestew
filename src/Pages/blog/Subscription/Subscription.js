import React from "react";
import "./Subscription.scss";

const Subscription = () => {
  return (
    <div className="subscription my-5 p-5">
      <div className="row">
        <div className="content text-lg-start text-center col-12 col-lg-6">
          <h3>Want to get weekly tips and tricks?</h3>
          <p role="paragraph">Sign up and receive your first email guide!</p>
        </div>
        <div className="sub align-items-center d-block d-lg-flex col-12 text-lg-end text-center col-lg-6">
          <input
            className="mb-4 mb-lg-0"
            type="email"
            name="email"
            placeholder="Your Email"
          />
          <a href="#0">Subscribe</a>
        </div>
      </div>
    </div>
  );
};

export default Subscription;
