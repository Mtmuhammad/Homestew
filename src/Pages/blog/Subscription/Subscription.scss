@use "../../../Styles/abstracts/mixins" as *;
@use "../../../Styles/abstracts/variables" as *;

.subscription {
  background-color: $light-primary-green;
  border-radius: 5px;

  & input {
    outline: none;
    border-radius: 5px;
    width: 75%;
    font-size: 1.6rem;
    letter-spacing: 0.1em;
    padding: 1.2rem;
  }

  & p {
    font-size: 1.8rem;
  }

  & a {
    color: #fff;
    background-color: $primary-green;
    padding: 1.2rem;
    font-size: 1.7rem;
    text-transform: uppercase;
    border-radius: 5px;
    transition: all 0.3s;

    @include respond(port) {
      display: block;
      width: 30%;
      margin: 0 auto;
    }

    &:hover {
      color: $primary-font;
    }
  }

  & h3 {
    font-size: 2.8rem;
  }
}
