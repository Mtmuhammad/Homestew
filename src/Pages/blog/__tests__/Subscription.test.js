import React from "react";
import { render, screen } from "@testing-library/react";
import Subscription from "../Subscription/Subscription";

describe("smoke and snapshot tests", () => {
  test("should render without crashing", () => {
    render(<Subscription />);
  });

  test("should match snapshot", () => {
    const { asFragment } = render(<Subscription />);
    expect(asFragment()).toMatchSnapshot();
  });
});

test("should render heading", () => {
  render(<Subscription />);
  const heading = screen.getByRole("heading");
  expect(heading).toBeInTheDocument();
  expect(heading).toHaveTextContent(/Want to get weekly tips and tricks?/i);
});
test("should render paragraph", () => {
  render(<Subscription />);
  const paragraph = screen.getByRole("paragraph");
  expect(paragraph).toBeInTheDocument();
  expect(paragraph).toHaveTextContent(
    /Sign up and receive your first email guide!/i
  );
});

test("should render input", () => {
  render(<Subscription />);
  const input = screen.getByPlaceholderText(/your email/i);
  expect(input).toBeInTheDocument();
  expect(input).toHaveClass("mb-4");
});
