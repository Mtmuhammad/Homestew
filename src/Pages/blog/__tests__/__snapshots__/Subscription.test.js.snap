// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`smoke and snapshot tests should match snapshot 1`] = `
<DocumentFragment>
  <div
    class="subscription my-5 p-5"
  >
    <div
      class="row"
    >
      <div
        class="content text-lg-start text-center col-12 col-lg-6"
      >
        <h3>
          Want to get weekly tips and tricks?
        </h3>
        <p
          role="paragraph"
        >
          Sign up and receive your first email guide!
        </p>
      </div>
      <div
        class="sub align-items-center d-block d-lg-flex col-12 text-lg-end text-center col-lg-6"
      >
        <input
          class="mb-4 mb-lg-0"
          name="email"
          placeholder="Your Email"
          type="email"
        />
        <a
          href="#0"
        >
          Subscribe
        </a>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
